### ISSUE TITLE
r-bookdown

### ISSUE THREAD

**Issue by CJ-Wright** [regro/cf-scripts#73]
There seems to be something problematic with `r-bookdown` It keeps causing `03` to error.
I'm not certain what is going on here, the script should be insulated from these kinds of issues. But it seems that doctr is getting an error.

If I had to guess what is going wrong something is having issues with `indir` where we are not popping the dir correctly.

**Comment by CJ-Wright** [comment]
@isuruf It seems that not all the versions in https://cran.r-project.org/src/contrib/ are in the associated archive (https://cran.r-project.org/src/contrib/Archive/mclust/ for example) should I pull from the `contrib` url without the archive?

**Comment by isuruf** [comment]
Sure.

**Comment by asmeurer** [comment]
Is there some bug with doctr here? Where is the failing build log?

**Comment by CJ-Wright** [comment]
The bug isn't with doctr, it was elsewhere. I can try to find the log, but I think I got the problem there. The inner loop usually cleans up after itself finishing the bump, but since it didn't do that doctr added the file to the repo, but now it seems to be doing the correct folder removal.

**Comment by asmeurer** [comment]
OK. I have an email filter for the word "doctr" so just checking. Open a bug in the doctr tracker if you do determine there is a doctr issue somewhere.

**Comment by CJ-Wright** [comment]
ah, very sorry. I'll make a point to throw an escape character in!

**Comment by CJ-Wright** [comment]
I think this issue is closed.

The multi source nature of the R packages caused some headaches in `03` (since string methods don't really work on lists). This caused an error cascade due to a bad access to the `meta.yaml` dict in the exception handling. This then seemed to cause issues with `indir` context manager, causing the directories to not be properly removed or the working directory not to be popped back to the top level. This caused the very helpful project-that-must-not-be-named to error as it could not commit a deleted dir.

These issues were fixed in the graph repo by removing the offending dir.

Additional fixes deal with the multi source R configuration (with R PRs now being sent) and the bad dict access was cleaned up. Additionally exceptions which propagate to the top of the script are now stashed into an `md` file for eventual reading.

Long term we should most likely bake some of this information (eg. the upstream urls to get hashes from in the source classes from `02` (if they aren't already there) and use them in a proper library, see: #41)

Most likely we should purge our various error trackers (`bad.txt`, `upstream_bad` and `exceptions.*`) so we can start fresh since we now support many more configurations then we did before.

Very sorry for the outage everyone.
